superset-frontend/src/components/Form/LabeledErrorBoundInput.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { Input } from 'antd';
+import { styled, css, SupersetTheme } from '@superset-ui/core';
+import FormItem from './FormItem';
+import FormLabel from './FormLabel';
+
+export interface LabeledErrorBoundInputProps {
+  label?: string;
+  name: string;
+  validationMethods:
+    | { onBlur: (value: any) => void }
+    | { onChange: (value: any) => void };
+  errorMessage: string | null;
+  helpText?: string;
+  value: string | number | readonly string[] | undefined;
+  required?: boolean;
+  placeholder?: string;
+  autoComplete?: string;
+  type?: string;
+  id?: string;
+  onChange?: any;
+}
+
+const StyledInput = styled(Input)`
+  margin: 8px 0;
+`;
+
+const alertIconStyles = (theme: SupersetTheme, hasError: boolean) => css`
+  .ant-form-item-children-icon {
+    display: none;
+  }
+  ${hasError &&
+  `.ant-form-item-control-input-content {
+      position: relative;
+
+      &:after {
+        content: ' ';
+        display: inline-block;
+        background: ${theme.colors.error.base};
+        mask: url('/images/icons/error.svg');
+        mask-size: cover;
+        width: ${theme.gridUnit * 4}px;
+        height: ${theme.gridUnit * 4}px;
+        position: absolute;
+        right: 7px;
+        top: 15px;
+      }
+    }`}
+`;
+
+const LabeledErrorBoundInput = ({
+  label,
+  validationMethods,
+  errorMessage,
+  helpText,
+  required = false,
+  id,
+  ...props
+}: LabeledErrorBoundInputProps) => (
+  <>
+    <FormLabel htmlFor={id} required={required}>
+      {label}
+    </FormLabel>
+    <FormItem
+      css={(theme: SupersetTheme) => alertIconStyles(theme, !!errorMessage)}
+      validateTrigger={Object.keys(validationMethods)}
+      validateStatus={errorMessage ? 'error' : 'success'}
+      help={errorMessage || helpText}
+      hasFeedback={!!errorMessage}
+      {...props}
```

Comment:
remind me, did we say that we were going to spread the props into the input first and then let people take them out explicitly if they needed them here?

Reply:
Oh you're right! Fixed it [`here`](https://github.com/apache/superset/pull/14530/commits/f476e27f08885c461b448c6b6042f5d149ec1935#diff-97dbad737572defaaa9f77fd37e305d4ef4080680a20708c7018dac7c47161b0L88)